static/src/js/containers/AuthCallbackContainer/AuthCallbackContainer.js

```diff
@@ -39,42 +42,40 @@ export const AuthCallbackContainer = ({
       redirect = '/'
     } = params
 
-    // Verify that the redirect params are real URLs
-    try {
-      let redirectUrl
-      if (eddRedirect) redirectUrl = new URL(eddRedirect)
-      if (redirect && redirect !== '/') redirectUrl = new URL(redirect)
-
-      if (
-        redirectUrl
-        && redirectUrl.protocol !== 'http:'
-        && redirectUrl.protocol !== 'https:'
-        && redirectUrl.protocol !== 'earthdata-download:'
-      ) {
-        // The redirectUrl is not a valid protocol
-        console.log('The redirectUrl is not a valid protocol')
-        window.location.replace('/')
+    let eddRedirectUrl = eddRedirect
+    if (redirect.includes('earthdata-download')) {
+      eddRedirectUrl = redirect
+    }
+
+    // Handle EDD redirects
+    if (eddRedirectUrl) {
+      const validEddRedirect = eddRedirectUrl.startsWith('earthdata-download')
+      if (validEddRedirect) {
+        if (accessToken) eddRedirectUrl += `&token=${accessToken}`
+
+        // Add the redirect information to the store
+        onAddEarthdataDownloadRedirect({
+          redirect: eddRedirectUrl
+        })
+
+        // Redirect to the edd callback
+        history.push('/earthdata-download-callback')
 
         return
       }
-    } catch (error) {
+
       window.location.replace('/')
 
       return
     }
 
-    // If the redirect includes earthdata-download, redirect to the edd callback
-    if (eddRedirect || redirect.includes('earthdata-download')) {
-      let eddRedirectUrl = eddRedirect || redirect
-      if (accessToken) eddRedirectUrl += `&token=${accessToken}`
+    // Handle redirects
+    const invalidRedirectUrl = redirect !== '/' && !redirect.startsWith(edscHost)
 
-      // Add the redirect information to the store
-      onAddEarthdataDownloadRedirect({
-        redirect: eddRedirectUrl
-      })
-
-      // Redirect to the edd callback
-      history.push('/earthdata-download-callback')
+    if (invalidRedirectUrl) {
+      // Redirect to an error page or a safe location if the URL is not a relative path
+      // https://developer.mozilla.org/en-US/docs/Web/API/Location/replace assign prevents back-button use in history
+      window.location.replace('/')
```

Comment:
It seems like we could send the user to the standard error page here. I _think_ a "replace" makes more sense than a "assign" here.